lib/auth/in_memory/builder.js

```diff
@@ -4,23 +4,23 @@ const serviceAccountPrefix =
 /** build simple authdata with only one account
  * @param {string} accessKey - account's accessKey
  * @param {string} secretKey - account's secretKey
+ * @param {string} canonicalId - account's canonical id
  * @param {string} [serviceName] - service name to use to generate can id
  * @return {object} authdata - authdata with account's accessKey and secretKey
  */
-function buildAuthDataAccount(accessKey, secretKey, serviceName) {
+function buildAuthDataAccount(accessKey, secretKey, canonicalId, serviceName) {
```

Comment:
Now that service account canonical ids contain the service type, we can remove the `serviceName` handling altogether.

Reply:
I kept `serviceName` for legacy purpose. If the canonicalId is not specified `buildAuthDataAccount` create a canonical id as: `${serviceAccountPrefix}/${serviceName}`